Tests/PublisherTests/DefaultPublisher/DefaultPublisherTests.swift

```diff
@@ -21,17 +25,28 @@ class DefaultPublisherTests: XCTestCase {
             return handler
         }
     }
-
+    
     override func setUpWithError() throws {
         locationService = MockLocationService()
         ablyService = MockAblyPublisherService()
-        configuration = ConnectionConfiguration(apiKey: "API_KEY", clientId: "CLIENT_ID")
         mapboxConfiguration = MapboxConfiguration(mapboxKey: "MAPBOX_ACCESS_TOKEN")
         resolutionPolicyFactory = MockResolutionPolicyFactory()
         routeProvider = MockRouteProvider()
         trackable = Trackable(id: "TrackableId",
                               metadata: "TrackableMetadata",
                               destination: CLLocationCoordinate2D(latitude: 3.1415, longitude: 2.7182))
+        try setUpUsingAPIKeyWithError()
+    }
+    
+    override func tearDownWithError() throws {
+        configuration = nil
+    }
+    
+    func setUpUsingAPIKeyWithError() throws {
+        if (configuration != nil) {
+            throw ClientConfigError.redefinedConnectionConfiguration
```

Comment:
This is not that readable. Maybe we should add associated value (string) with message, and throw it together? For example:

`throw ClientConfigError.redefinedConnectionConfiguration(message: "The client connection configuration has already been defined.")`

My error message is only a guess, because I am not sure what it's doing there.

Reply:
After checking the git blame, turns out I wrote that. 😂  20e6c87300de79241f8239fd0d0fd369b60713d7 renames this error enum to be clearer.